AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderSigninTests.swift

```diff
@@ -39,7 +50,716 @@ class AuthenticationProviderSigninTests: BaseAuthenticationProviderTest {
+    ///    - I invoke signIn
+    /// - Then:
+    ///    - I should get a .confirmSignInWithSMSMFACode
+    ///
+    func testSignInWithNextStepSMS() {
+
+        let mockSigninResult = SignInResult(signInState: .smsMFA)
+        mockAWSMobileClient?.signInMockResult = .success(mockSigninResult)
+
+        let options = AuthSignInRequest.Options()
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signIn(username: "username", password: "password", options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                guard case .confirmSignInWithSMSMFACode = signinResult.nextStep else {
+                    XCTFail("Result should be .confirmSignInWithSMSMFACode for next step")
+                    return
+                }
+                XCTAssertFalse(signinResult.isSignedIn, "Signin result should not be complete")
+            case .failure(let error):
+                XCTFail("Received failure with error \(error)")
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with customChallenge as signIn result response
+    ///
+    /// - Given: Given an auth plugin with mocked service. Mock customChallenge response for signIn result
+    ///
+    /// - When:
+    ///    - I invoke signIn
+    /// - Then:
+    ///    - I should get a .confirmSignInWithCustomChallenge
+    ///
+    func testSignInWithNextStepCustomChallenge() {
+
+        let mockSigninResult = SignInResult(signInState: .customChallenge)
+        mockAWSMobileClient?.signInMockResult = .success(mockSigninResult)
+
+        let options = AuthSignInRequest.Options()
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signIn(username: "username", password: "password", options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                guard case .confirmSignInWithCustomChallenge = signinResult.nextStep else {
+                    XCTFail("Result should be .confirmSignInWithCustomChallenge for next step")
+                    return
+                }
+                XCTAssertFalse(signinResult.isSignedIn, "Signin result should not be complete")
+            case .failure(let error):
+                XCTFail("Received failure with error \(error)")
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with newPassword as signIn result response
+    ///
+    /// - Given: Given an auth plugin with mocked service. Mock newPassword response for signIn result
+    ///
+    /// - When:
+    ///    - I invoke signIn
+    /// - Then:
+    ///    - I should get a .confirmSignInWithNewPassword
```

Comment:
nit: I should get a . confirmSignInWithNewPassword error?

Reply:
Fixed